mcstatus/__main__.py

```diff
@@ -1,75 +1,103 @@
 from __future__ import annotations
 
+import sys
 import argparse
 import socket
+from typing import TYPE_CHECKING
 from json import dumps as json_dumps
 
-from mcstatus import JavaServer
+from mcstatus import JavaServer, BedrockServer
+from mcstatus.responses import JavaStatusResponse
 
+if TYPE_CHECKING:
+    SupportedServers = JavaServer | BedrockServer
 
-def ping(server: JavaServer) -> None:
-    print(f"{server.ping()}ms")
+def ping(server: SupportedServers) -> int:
+    print(f"{server.status().latency}")  # XXX: JavaServer.ping() not working?
```

Comment:
Will 850 be fixed in a way that `server.ping()` works for all servers? If not, I'd consider the CLI an external application using the mcstatus library and implementing the suggested `status().latency` workaround.

Reply:
Looking at my last comment it will, but if not, I would prefer
```py
try:
    return server.ping()
except Exception:
    warnings.warning(...)
    return server.status().latency

Reply:
Can we leave it as `server.status().latency` for now, waiting for the 850 resolution to see what to change it to? Bedrock also doesn't have ping(), so it will require `status().latency` regardless.

Reply:
I would happily create a fix for #850, but right now I cannot come up with a good explanation in docs. Feel free to submit a PR

(`git apply patch.diff`)
https://pastebin.com/F8Q9aVQh

Reply:
That patch doesn't make ping() functional for all servers? I'll make a change to use your try/catch suggestion.

Reply:
@PerchunPak If you'd like to open your current patch into a PR, it would be better than this thread, I think. My first feedback would be to make the warning more prolific, including in docstrings of the ping() method.

Reply:
> That patch doesn't make ping() functional for all (Java) servers? I'll make a change to use your try/catch suggestion.

I thought everyone agreed to just document the bug and do nothing else?

> My first feedback would be to make the warning more prolific, including in docstrings of the ping() method.

Yeah, that's the problem. I just don't know what to write, it must be more informative though

Reply:
I thought the topic of 850 was limited to the ping() API method. Since the CLI is a separate thing and it is used by (possibly non-technical) users, I thought it should use the status().latency workaround itself so it "just works" ([as mentioned](https://github.com/py-mine/mcstatus/pull/849#discussion_r1680880840)). Is this reasonable?

Reply:
Not really, `ping` is much lighter than `status`, so we (or user) should prefer `ping` over `status`, but if it doesn't work - `status().latency` is the workaround